src/pollen.py

```diff
@@ -4,57 +4,49 @@
 """Pollen charm business logic."""
 
 import glob
-import os
-import shutil
 import subprocess
-import time
+from pathlib import Path
 
 from charms.operator_libs_linux.v0 import apt
 from charms.operator_libs_linux.v1 import systemd
+from charms.operator_libs_linux.v2 import snap
 
 from exceptions import ConfigurationWriteError, InstallError
 
-POLLEN_SERVICE_NAME = "pollen.service"
-
 
 class PollenService:
     """Pollen service class."""
 
     @classmethod
-    def prepare(cls) -> None:
+    def prepare(cls, unit_name) -> None:
         """Install packages and write configuration files.
 
+        Args:
+            unit_name: ¨Pollen charm's unit name.
+
         Raises:
             InstallError: if the packages fail to install
             ConfigurationWriteError: something went wrong writing the configuration
         """
         try:
             apt.update()
-            apt.add_package(["pollen", "pollinate", "ent"])
-            # Apt-get makes the pollen restart error if we don't wait,
-            # even if apt-get executes correctly and there are no
-            # child processes left. The cause is unknown.
-            time.sleep(10)
+            apt.add_package(["pollinate", "ent"])
+            snap.add("gtrkiller-pollen", channel="candidate")
```

Comment:
Should this have your handle in the name of the snap?

Reply:
The snap hasn't been published yet, so for now it will have my name in it until that happens and the ownership discussion ends.